idea_town/frontend/static-src/app/main.js

```diff
@@ -18,32 +20,50 @@ app.extend({
     fetch('/api/me?format=json', {
       credentials: 'same-origin'
     }).then((response) => response.json()).then((userData) => {
+      app.on('webChannel:addon-available', () => {
+        app.me.hasAddon = true;
+        app.experiments = new ExperimentsCollection();
+        app.experiments.fetch({
+          success: app.blastOff
+        });
+      });
       app.me = new Me({
         user: userData
       });
+    }).catch((err) => {
+      // for now, log the error in the console & do nothing in the UI
+      console && console.error(err); // eslint-disable-line no-console
+    });
+  },
 
-      app.experiments = new ExperimentsCollection();
-      app.experiments.fetch();
-
-      // session won't change without a hard refresh, but addon state could, so:
-      // if addon state changes, dump user back to '/' and let the router handle
-      // redirecting to the correct landing page
-      app.me.on('change:hasAddon', () => { app.router.reload(); });
+  blastOff() {
```

Comment:
Hmm, we get here after 1) the addon signals availability and 2) the experiments have loaded.

But, if the addon isn't installed, it never signals availability. And so experiments never load. And so we never end up here to initialize the PageManager & Router. That seems like a problem.